src/TriggerActivityMakerTrackFinding.cpp

```diff
@@ -0,0 +1,303 @@
+      win_adj.add(chanTPList[next].second);
+      for (int i = 0; i < next_channel-channel-1; ++i) ++tol_count;
+    }
+
+    // if track length > m_adjacency_threshold, set win_adj_max = win_adj;
+    else if (win_adj.inputs.size() > m_adjacency_threshold) {
+      win_adj_max = win_adj;
+      break;
+    }
+    
+    // If track length < m_adjacency_threshold, reset variables for next iteration.
+    else {
+      tol_count = 0;
+      win_adj.clear();
+    }
+  }
+  
+  return win_adj_max;
+}
+
+// =====================================================================================
+// Functions below this line are for debugging purposes.
+// =====================================================================================
+void
+TriggerActivityMakerTrackFinding::add_window_to_record(TPWindow window)
+{
+  m_window_record.push_back(window);
+  return;
+}
+
+// Function to dump the details of the TA window currently on record
+void
+TriggerActivityMakerTrackFinding::dump_window_record()
+{
+  std::ofstream outfile;
+  outfile.open("window_record_tam.csv", std::ios_base::app);
+
+  for (auto window : m_window_record) {
+    outfile << window.time_start << ",";
+    outfile << window.inputs.back().time_start << ",";
+    outfile << window.inputs.back().time_start - window.time_start << ",";
+    outfile << window.adc_integral << ",";
+    outfile << window.n_channels_hit() << ",";       // Number of unique channels with hits
+    outfile << window.inputs.size() << ",";          // Number of TPs in TPWindow
+    outfile << window.inputs.back().channel << ",";  // Last TP Channel ID
+    outfile << window.inputs.front().channel << ","; // First TP Channel ID
+    outfile << check_tot() << std::endl;             // Summed window TOT
+  }
+
+  outfile.close();
+  m_window_record.clear();
+
+  return;
+}
+
+// Function to add current TP details to a text file for testing and debugging.
+void
+TriggerActivityMakerTrackFinding::dump_tp(TriggerPrimitive const& input_tp)
+{
+  std::ofstream outfile;
+  outfile.open("coldbox_tps.txt", std::ios_base::app);
+
+  // Output relevant TP information to file
+  outfile << input_tp.time_start << " ";          
+  outfile << input_tp.time_over_threshold << " "; // 50MHz ticks
+  outfile << input_tp.time_peak << " ";           
+  outfile << input_tp.channel << " ";             // Offline channel ID
+  outfile << input_tp.adc_integral << " ";        
+  outfile << input_tp.adc_peak << " ";            
+  outfile << input_tp.detid << " ";               // Det ID - Identifies detector element
+  outfile << input_tp.type << std::endl;        
+  outfile.close();
+
+  return;
+}
+
+int
+TriggerActivityMakerTrackFinding::check_tot() const
+{
+  // Here, we just want to sum up all the tot values for each TP within window,
+  // and return this tot of the window.
+  int window_tot = 0; 
+  for (auto tp : m_current_window.inputs) {
+    window_tot += tp.time_over_threshold;
+  }
+
+  return window_tot;
+}
```

Comment:
Only used in `dump_window_record`, which is not used anywhere, could remove.

Reply:
done